### DISCUSSION TITLE
Model for stock splits

### DISCUSSION THREAD

**Discussion by kedder** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#103] [Suggestions] [answered]
Currently, stock splits are modeled as a series of transactions on security objects, e.g. [Warrant split](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/main/docs/schema/objects/transactions/split/schema-objects-transactions-split-warrant_split.md), [Stock Split](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/main/docs/schema/objects/transactions/split/schema-objects-transactions-split-stock_split.md) or [Plan Security Split](
https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/main/docs/schema/objects/transactions/split/schema-objects-transactions-split-plan_security_split.md).

This model seems to suggest that same-stock class securities can be split independently (e.g. option grant A with one split ratio, option grant B with a different one). However, this shouldn't happen in practice. All securities of the same stock class are split the same way and at the same moment in time. Otherwise capitalization can be affected in an undesired way - if split transaction dates or rations are inconsistent, split transactions can skew the ownership percentages, which goes against the definition of a stock split.

Consider modeling stock splits as a dedicated “objects” (containing at least data, stock class and ratio) and refer to it in individual warrant/stock/option grant split transactions.

**Comment by JSv4** [comment]
@kedder, we've definitely considered this approach. If you or someone on your team has the bandwidth to spec out what this might look like, would you be open to taking a stab at modeling this?

**Reply by kedder** [reply]
@JSv4 sure, I can try. Would you prefer a PR for data schemas or more detailed description here?

**Reply by kedder** [reply]
@JSv4 I have a (local) branch with suggested changes for stock splits. However I can't neither fork nor push a branch to this repo. How can I propose the change?

**Reply by JSv4** [reply]
Awesome, @kedder! I temporarily elevated your access to so you can create a branch. Thanks for taking the time to contribute! We'll take a look at what you put together.

**Reply by kedder** [reply]
Hm, still don't seem to be able to do that: 

```
$ git push --set-upstream origin kedder/splits
ERROR: Write access to repository not granted.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Reply by JSv4** [reply]
@kedder, give it another shot.

**Reply by kedder** [reply]
It worked, thanks!

**Reply by kedder** [reply]
Submitted #116.

**Reply by JSv4** [reply]
Thanks! Looks like a really solid approach. I'll raise in the next Technical Working Group call this week for discussion.

**Comment by JSv4** [comment, marked as answer]
These changes are being incorporated via PR #116 submitted by @kedder.
